### ISSUE TITLE
"failed retrieving Azure location for account" - error message is showing wrong (personal) Subscription ID instead of ID used for login (work)

### ISSUE THREAD

**Issue by azaslawski** [Azure/azure-dev#4379]
### Describe the bug:
Logged in with `azd auth login` - with work account
Executed `azd up` to deploy my Bicep script on the above account.
Got Below Error  problem is that the message is showing my personal subscription ID, not work ID with which I have logged in (redacted PII):
> ERROR: listing locations: failed retrieving Azure location for account <displaying_personal_acct_id_INCORRECT_should_be_work_ID>: failed to resolve user <displaying_work_user_acct_CORRECT> access to subscription with ID <displaying_personal_acct_id_INCORRECT_should_be_work_ID> If you recently gained access to this subscription, run `azd auth login` again to reload subscriptions.
> Otherwise, visit this subscription in Azure Portal using the browser, then run `azd auth login` 

### Used commands:
`azd auth login` - login process completed w/o issues
`azd up` - getting above error

### Issue script & Debug output
executed with debugging on:
`azd up --debug -> log_azd.txt`
file content:

> Initialize bicep provider
> Retrieving locations...
> 
> ERROR: listing locations: failed retrieving Azure location for account <displaying_personal_acct_id_INCORRECT_should_be_work_ID>: failed to resolve user <displaying_work_user_acct_CORRECT> access to subscription with ID <displaying_personal_acct_id_INCORRECT_should_be_work_ID> If you recently gained access to this subscription, run azd auth login again to reload subscriptions.
> Otherwise, visit this subscription in Azure Portal using the browser, then run azd auth login

### Expected behavior:
after executing `azd up` I expect a location prompt to come up so that I can select a location: i.e. eastus, instead the above error is displaying.

### Environment Summary:
azd version 1.10.1 (commit 31409a33266fb4a5fdbb644bc83988e725d6c7c9)
azure-cli 2.64.0
core 2.64.0
telemetry 1.1.0
Dependencies:
msal 1.30.0
azure-mgmt-resource 23.1.1
Python location '/opt/az/bin/python3'
Extensions directory '/home/<username>/.azure/cliextensions'
Python (Linux) 3.11.8 (main, Aug 28 2024, 05:45:18) [GCC 11.4.0]
Legal docs and information: aka.ms/AzureCliLegal
Your CLI is up-to-date.

### Additional context:
I have two VMs running Ubuntu operating systems 22.04.4 LTS on top on VirtualBox.
I use both VMs for development, but each is using a different Azure account (personal, and work) nothing is shared between these VMs, other than they run on the same OS (Win11-23H2, 12th Gen i7-12700K 3.60 GHz, 128 GB Ram). Work VM is on a Bridged network and Personal VM is on NAT.
When executing 'azd up' on VM1 (personal Azure account), bicep script runs w/o issues and generates resources.
BUT
Wen executing 'azd up' on VM2 (work Azure account), bicep script does not work and I get an error (specified below) also even though the error message shows my correct (work) user id, it (somehow) is displaying my personal subscription id.
Also, when setting up 'azd init' it correctly shows my Work Azure Subscriptions at which point a correct subscription is selected.

**Comment by rajeshkamal5050** [comment]
@azaslawski thanks for reporting.

Can you check this issue - https://github.com/Azure/azure-dev/issues/2640#issuecomment-1684378415 if similar?

cc @weikanglim @vhvb1989

**Comment by azaslawski** [comment]
Yes the last comment in the issue: https://github.com/Azure/azure-dev/issues/2640#issuecomment-1684378415  aligns with the issue I have reported.
Ultimately I have taken a snapshot of my VM and deleted subscription.cache as well bin/msal/cache.json inside .azd directory (although looking at its content I didn't see any anomalies)
Also I have deleted my environment:  .azure/test/.env file content.
Re-run standard logon process and 'azd up' is working now, and ARM finished creating resources as per bicep script.
